File: problems/Python/128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        mx = 0
        
        for n in s:
            if n - 1 not in s:
                cur = n
                l = 1
                while cur + 1 in s:
                    cur += 1
                    l += 1
                if l > mx:
                    mx = l
        return mx

File: problems/Java/128. Longest Consecutive Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> s = new HashSet<>();
        for (int n : nums) s.add(n);
        
        int mx = 0;
        for (int n : s) {
            if (!s.contains(n - 1)) {
                int cur = n;
                int l = 1;
                while (s.contains(cur + 1)) {
                    cur++;
                    l++;
                }
                if (l > mx) mx = l;
            }
        }
        return mx;
    }
}